device/kokkos/src/seeding/seed_finding.cpp

```diff
@@ -0,0 +1,193 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021-2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Local include(s).
+#include "traccc/kokkos/seeding/seed_finding.hpp"
+#include "traccc/cuda/utils/definitions.hpp"
+
+// Library include(s).
+#include "traccc/kokkos/seeding/seed_selecting.hpp"
+#include "traccc/kokkos/seeding/triplet_counting.hpp"
+#include "traccc/kokkos/seeding/triplet_finding.hpp"
+#include "traccc/kokkos/seeding/weight_updating.hpp"
+
+// Project include(s).
+#include "traccc/device/get_prefix_sum.hpp"
+#include "traccc/edm/device/doublet_counter.hpp"
+#include "traccc/seeding/device/count_doublets.hpp"
+#include "traccc/seeding/device/find_doublets.hpp"
+#include "traccc/seeding/device/make_doublet_buffers.hpp"
+#include "traccc/seeding/device/make_doublet_counter_buffer.hpp"
+
+// VecMem include(s).
+#include "vecmem/utils/copy.hpp"
+
+// Kokkos include(s).
+#include <Kokkos_Core.hpp>
+
+// System include(s).
+#include <algorithm>
+#include <vector>
+
+ 
+
+namespace traccc::kokkos {
+
+// defining MemSpace to generalize in case they use other
+#ifdef KOKKOS_ENABLE_CUDA
+#define MemSpace Kokkos::CudaSpace
+#endif
+    
+#ifndef MemSpace
+#define MemSpace Kokkos::HostSpace
+#endif 
+
+// defining execution space and range_policy
+using ExecSpace = Memspace::execution_space;
+using range_policy = Kokkos::RangePolicy<ExecSpace>; 
+
+typedef Kokkos::TeamPolicy<ExecSpace> team_policy;
+typedef Kokkos::TeamPolicy<ExecSpace>::member_type member_type;
+
+
+seed_finding::seed_finding(const seedfinder_config& config,
+                           vecmem::memory_resource& mr)
+    : m_seedfinder_config(config), m_mr(mr) {}
+
+seed_finding::output_type seed_finding::operator()(
+    const spacepoint_container_types::host& spacepoints,
+    const sp_grid_const_view& g2_view) const {
+
+    // Helper object for the data management.
+    vecmem::copy copy;
+
+    // Get the prefix sum for the spacepoint grid.
+    const device::prefix_sum_t sp_grid_prefix_sum =
+        device::get_prefix_sum(g2_view._data_view, m_mr.get(), copy);
+
+    // Set up the doublet counter buffer.
+    device::doublet_counter_container_types::buffer doublet_counter_buffer =
+        device::make_doublet_counter_buffer(g2_view, copy, m_mr.get());
+// new code
+    // initilize Kokkos 
+    Kokkos::initialize( argc, argc );
```

Comment:
The Kokkos initialisation and finalisation must be moved into the executable that will be using this algorithm / these algorithms. It must not be done inside of the algorithm itself.